reana_job_controller/htcondorcern_job_manager.py

```diff
@@ -162,33 +162,27 @@ def _replace_absolute_paths_with_relative(self, cmd):
         return relative_paths_command
 
     def _format_arguments(self):
-        r"""Format HTCondor job execution arguments.
-
-        input  - ['bash', '-c',
-                  'cd /var/reana/users/00000000-0000-0000-0000-000000000000/
-                       workflows/e4691f78-25aa-4f90-9c3d-6873a97bdf16 ;
-                   python "code/helloworld.py" --inputfile "data/names.txt"
-                           --outputfile "results/greetings.txt" --sleeptime 0']
-        output - python \"code/helloworld.py\" --inputfile \"data/names.txt\"
-                 --outputfile \"results/greetings.txt\" --sleeptime 0
-        """
+        """Format HTCondor job execution arguments."""
         if self.workflow.type_ == "serial":
-            base_cmd = " ".join(self.cmd[2].split()[3:])
+            # Take only the user's command, removes the change directory to workflow workspace
+            # added by RWE-Serial since HTCondor implementation does not need it.
+            # E.g. "cd /path/to/workspace ; user-command" -> "user-command"
+            base_cmd = " ".join(self.cmd.split()[3:])
```

Comment:
This implementation has been simplified but not fully refactored and the reason is that RWE-Serial needs jobs to run on the workflow workspace. However, this is not something that can be decided by RJC since not all workflow engines support/require jobs to work on the workflow workspace.

We should split the two entities, workon workflow workspace and command. Then add a new field in the `POST /jobs` endpoint, e.g. `workon_workflow_workspace`, to specify whether the job should be run inside the workflow workspace. Then, each compute backend implementation should decide how to implement it.